operator/controllers/seldondeployment_controller.go

```diff
@@ -1072,6 +1072,18 @@ func (r *SeldonDeploymentReconciler) createIstioServices(components *components,
 		}
 	}
 
+	if ready {
+		var reason string
+		if len(components.virtualServices) > 0 {
```

Comment:
For this wouldn't we need to add the exact number of services, just in case for example if there's a new canary/shadow added and the virtualservice needs to be created for it